### ISSUE TITLE
add warning if common.states is not type object

### ISSUE THREAD

**Issue by Garfonso** [ioBroker/ioBroker.js-controller#1389]
Please add another "invalid object" warning if common.states exists and is not of type `object` (but string). 

From the docs, only objects like {1: "xy", 2: "z"} and (new) arrays of type ["xy", "z"] should be allowed. But there are a lot of objects that encode the states in a single string, like "1:xy;2:z" and similar which is very tedious to support... not sure if this works in admin... but I'd vote for a warning for such states in order to try to fade out this kind of states conversion.

**Comment by foxriver76** [comment]
Not sure how many adapters will be affected by this, but I guess not many so I would support this change. What do you think @Apollon77

**Comment by Apollon77** [comment]
@GermanBluefox What is the supposed way? But yes officially supported is the "object with only numeric keys" way

**Comment by foxriver76** [comment]
With three different possibilities I can imagine that it is annoying for e.g. visualization developers like @Garfonso

**Comment by Garfonso** [comment]
As I mentioned, the official docs only allow the object with numeric keys (and I think we wanted to add string arrays, and we should do that if not already happened, IMHO). 

I don't know of any adapters that use the string version. But there seem to be a lot of scripts / self defined objects out there that use the string version. I implemented support for the string in all places I found ( 😄 ) but parsing a string always feels more brittle to me than interpreting contents of an object. For example it can not easily be enforced that a key is only used once. Also ; and : are illegal characters that can break the parsing in many ways, which I definitively do not handle in my implementations. 

Also the object way has the advantage that it can be displayed much more nicely in the JSON view whereas the string version becomes hard to read / edit with growing number of options.

So my vote goes for "deprecate it".
